### PR TITLE
Added recently searched images feature

### LINKED ISSUE #1569: Search Image History
**Summary:** 
This is an enhancement to #1521 (Search Images feature)
When the user touches the search icon in explore tab they see a blank screen. Instead, we can show the search history on the screen.

 **Commons app version:** 
2.7.1-debug_rename_featured-df82383

**Screen-shots:** 

screenshot_20180528-125309

**Would you like to work on the issue?**
Yes, I have already started working on this issue. I will send a PR soon for this feature.

### PR DESCRIPTION
**Pull request by ujjwalagrawal17**
## Added recently searched images feature

Fixes #1569 (Search Image History)

## Description 

- RecentSearchContentProvider added
- RecentSearchDao added
- Changed database version to 7.
- RecentSearchFragment added to show the list of recent searches 
- Managed visibility of recent search fragment and searchImageFragment according to query.

## Tests performed (required)

Manually Tested on API 25 & MOTO G5S+ , with {ProdDebug Variant}.

## Screenshots showing what changed 
<table>
<tr>
<td>

screenshot_20180605-110849

</td>
</tr>
</table>

### CODE REVIEW

**Comment by ujjwalagrawal17** [comment]
@nicolas-raoul  @neslihanturan  Can you test this feature and suggest improvements.

File: app/src/main/java/fr/free/nrw/commons/data/DBOpenHelper.java [hunk 1]
```diff
@@ -6,12 +6,13 @@
 
 import fr.free.nrw.commons.category.CategoryDao;
 import fr.free.nrw.commons.contributions.ContributionDao;
+import fr.free.nrw.commons.explore.recent_searches.RecentSearchesDao;
```
[the remark this answers is not in the document]
Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/data/DBOpenHelper.java]: Ok nicolas I will change it.

File: app/src/main/java/fr/free/nrw/commons/explore/recent_searches/RecentSearch.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import android.net.Uri;
+import java.util.Date;
+
+/**
+ * Represents a recently searched query
+ */
+public class RecentSearch {
+    private Uri contentUri;
+    private String name;
```
[the remark this answers is not in the document]
Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/explore/recent_searches/RecentSearch.java]: @nicolas-raoul Ok, I will refractor name to query.

**Comment by ujjwalagrawal17** [comment]
@nicolas-raoul @neslihanturan I have updated the PR. Can you review it again.

File: app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java [hunk 3]
```diff
@@ -123,7 +127,10 @@ private void handleSuccess(List<Media> mediaList) {
             progressBar.setVisibility(View.GONE);
             imagesAdapter.addAll(mediaList);
             imagesAdapter.notifyDataSetChanged();
-            saveQuery(query);
+
+            // check if user is waiting for 5 seconds if yes then save search query to history.
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: Great idea! :-)

Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: Thanks :)

File: app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java [hunk 4]
```diff
@@ -71,7 +72,10 @@ private void saveQuery(String query) {
         // Newly searched query...
         if (recentSearch == null) {
             recentSearch = new RecentSearch(null, query, new Date());
+        }else {
```
Comment (nicolas-raoul) [inline app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: Trivial but usually we write
```
}
else {
```
I think.

Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: Ok, @nicolas-raoul  I have updated it and will take care of it from next time.

**Comment by nicolas-raoul** [comment]
There is a conflict now, would you mind rebasing or pulling?

**Comment by ujjwalagrawal17** [comment]
@nicolas-raoul Doing it right now.

**Comment by ujjwalagrawal17** [comment]
@nicolas-raoul I have fixed the conflicts and tested the app again.

**Comment by neslihanturan** [comment]
Thanks for good job @ujjwalagrawal17 !

**Pull request merged**